game/shieldbattery/shieldbattery.cpp

```diff
@@ -384,13 +386,31 @@ DWORD __stdcall GetModuleFileNameAHook(
   }
 }
 
+HMODULE __stdcall LoadLibraryAHook(_In_ LPCTSTR filename) {
+  if (EndsWith(filename, "local.dll")) {
+    wchar_t starcraft_dir[MAX_PATH];
+    DWORD count = GetCurrentDirectoryW(sizeof(starcraft_dir), starcraft_dir);
```

Comment:
We have a path for this, its `current_dir_on_inject` or `real_starcraft_path`. You shouldn't rely on the current directory just happening to be the StarCraft path, I very particularly wrote this such that we save it off at the earliest possible time (before anything has an opportunity to change it). It is absolutely not okay to re-check it at this point in time.

This code is needlessly complex, really, you can just:
1) Check if they're loading local.dll
2) Call GetModuleFileNameW(NULL) [returns the filepath for starcraft.exe]
3) Replace the file name in that path with 'local.dll'
4) Return LoadLibraryW result of that path

There's only two cases here:
- We're using a downgrade version, in which case local.dll is beside it in the downgrade folder
- We're not using a downgrade version, in which case local.dll is beside it in the StarCraft folder

As you can see, in both cases, local.dll is always directly beside StarCraft.exe. If we didn't need the GetModuleFileNameA hook, hooking LoadLibrary here wouldn't be necessary at all, as the code that StarCraft runs would already try to load the right path.